### PR TITLE
Remove \Throwable bubbling down support deprecation layer

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Remove deprecation layer of https://github.com/symfony/symfony/pull/33065

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Can you please rebase to trigger the CI again?

File: src/Symfony/Component/HttpClient/DataCollector/HttpClientDataCollector.php [hunk 1]
```diff
@@ -18,8 +18,10 @@
 
 /**
  * @author Jérémy Romey <jeremy@free-agent.fr>
+ *
+ * @final
  */
-final class HttpClientDataCollector extends DataCollector
+class HttpClientDataCollector extends DataCollector
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpClient/DataCollector/HttpClientDataCollector.php]: No need to change this, that's how it is in 4.4, so it should stay as is.

File: src/Symfony/Bridge/Twig/composer.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
         "symfony/finder": "^4.4|^5.0",
         "symfony/form": "^5.0",
         "symfony/http-foundation": "^4.4|^5.0",
-        "symfony/http-kernel": "^4.4|^5.0",
+        "symfony/http-kernel": "^5.0",
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Twig/composer.json]: I think all similar changes should be reverted - DataCollectorInterface v4.4 is compatible with the new signatue.
Same for all components.

File: src/Symfony/Bridge/Twig/composer.json [hunk 3]
```diff
@@ -51,6 +51,7 @@
         "symfony/console": "<4.4",
         "symfony/form": "<5.0",
         "symfony/http-foundation": "<4.4",
+        "symfony/http-kernel": "<5.0",
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Twig/composer.json]: all conflicts need to be <4.4

File: src/Symfony/Component/HttpClient/DataCollector/HttpClientDataCollector.php (hunk 1 above)
Comment (fancyweb) [reply src/Symfony/Component/HttpClient/DataCollector/HttpClientDataCollector.php]: My bad, this one was not intentional 😃

File: src/Symfony/Component/HttpKernel/composer.json [hunk 4]
```diff
@@ -48,9 +48,18 @@
...
         "symfony/config": "<5.0",
+        "symfony/form": "<5.0",
         "symfony/dependency-injection": "<4.4",
-        "symfony/translation": "<4.4",
+        "symfony/doctrine-bridge": "<5.0",
+        "symfony/http-client": "<5.0",
+        "symfony/mailer": "<5.0",
+        "symfony/messenger": "<5.0",
+        "symfony/translation": "<5.0",
+        "symfony/security-bundle": "<5.0",
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/composer.json]: no needed

File: src/Symfony/Component/HttpKernel/composer.json [hunk 5]
```diff
@@ -48,9 +48,18 @@
     },
     "conflict": {
         "symfony/browser-kit": "<4.4",
+        "symfony/cache": "<5.0",
         "symfony/config": "<5.0",
+        "symfony/form": "<5.0",
         "symfony/dependency-injection": "<4.4",
-        "symfony/translation": "<4.4",
+        "symfony/doctrine-bridge": "<5.0",
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/composer.json]: can be removed

File: .travis.yml [hunk 6]
```diff
@@ -227,7 +227,7 @@ install:
           export SYMFONY_VERSION=$(git ls-remote --heads | grep -o '/[1-9].*' | tail -n 1 | sed s/.//) &&
           git fetch --depth=2 origin $SYMFONY_VERSION &&
           git checkout -m FETCH_HEAD &&
-          export COMPONENTS=$(find src/Symfony -mindepth 2 -type f -name phpunit.xml.dist -printf '%h\n' | sort)
+          export COMPONENTS=$(find src/Symfony -mindepth 2 -type f -name phpunit.xml.dist -not -wholename '*/Component/Debug/*' -printf '%h\n' | sort)
```
Comment (nicolas-grekas) [inline .travis.yml]: I think we can remove this - the failure that is triggered in relation here might be fixed by merging. If not, we'll apply.

File: src/Symfony/Component/HttpKernel/composer.json [hunk 7]
```diff
@@ -48,9 +48,18 @@
...
+        "symfony/http-client": "<5.0",
+        "symfony/mailer": "<5.0",
+        "symfony/messenger": "<5.0",
+        "symfony/translation": "<5.0",
+        "symfony/security-bundle": "<5.0",
+        "symfony/twig-bridge": "<5.0",
+        "symfony/validator": "<5.0",
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/composer.json]: then on 4.4, we should unallow `|^5.0` on all these excluded components.

**Review by nicolas-grekas — APPROVED** [review]
(once comments are addressed)

**Review by derrabus — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @fancyweb.

**Pull request merged**